src/Mono.Android/Java.Interop/TypeManager.cs

```diff
@@ -353,10 +365,18 @@ public static void RegisterType (string java_class, Type t)
 			}
 		}
 
-		static Dictionary<string, List<Converter<string, Type?>>> packageLookup = new Dictionary<string, List<Converter<string, Type?>>> ();
+		static Dictionary<string, List<Converter<string, Type?>>>? packageLookup;
+
+		static void LazyInitPackageLookup ()
+		{
+			if (packageLookup == null)
+				packageLookup = new Dictionary<string, List<Converter<string, Type?>>> ();
```

Comment:
I think the dictionary should use the `Ordinal` comparer (or `OrdinalIgnoreCase` if we expect the package name can differ in case) instead of the default culture-aware one.

Reply:
This was just refactored, shoudn't harm to use the comparer though, so updated.